aws/table_aws_vpc_verified_access_group.go

```diff
@@ -0,0 +1,237 @@
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.From(verifiedAccessGroupTitle),
+			},
+			{
+				Name:        "tags",
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.From(verifiedAccessGroupTurbotTags),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("VerifiedAccessGroupArn").Transform(transform.EnsureStringArray),
+			},
+		}),
+	}
+}
+
+//// LIST FUNCTION
+
+func listVpcVerifiedAccessGroups(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+
+	// Create session
+	svc, err := EC2Client(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_vpc_verified_access_group.listVpcVerifiedAccessGroups", "connection_error", err)
+		return nil, err
+	}
+
+	// Limiting the results
+	maxLimit := int32(200)
+	if d.QueryContext.Limit != nil {
+		limit := int32(*d.QueryContext.Limit)
+		if limit < maxLimit {
+			if limit < 5 {
+				maxLimit = 5
+			} else {
+				maxLimit = limit
+			}
+		}
+	}
+
+	input := &ec2.DescribeVerifiedAccessGroupsInput{
+		MaxResults: aws.Int32(maxLimit),
+	}
+
+	if d.KeyColumnQualString("verified_access_instance_id") != "" {
+		input.VerifiedAccessInstanceId = aws.String(d.KeyColumnQualString("verified_access_instance_id"))
+	}
+
+	for {
+		// List call
+		resp, err := svc.DescribeVerifiedAccessGroups(ctx, input)
+
+		if err != nil {
+			plugin.Logger(ctx).Error("aws_vpc_verified_access_group.listVpcVerifiedAccessGroups", "api_error", err)
+			return nil, nil
+		}
+
+		for _, group := range resp.VerifiedAccessGroups {
+			d.StreamListItem(ctx, group)
+
+			// Context may get cancelled due to manual cancellation or if the limit has been reached
+			if d.QueryStatus.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+		if resp.NextToken == nil {
+			break
+		} else {
+			input.NextToken = resp.NextToken
+		}
+	}
+
+	return nil, err
+}
+
+//// HYDRATED FUNCTION
+
+func getVpcVerifiedAccessGroup(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+
+	groupId := d.KeyColumnQuals["verified_access_group_id"].GetStringValue()
+
+	// Empty check
+	if groupId == "" {
+		return nil, nil
+	}
+
+	// Create session
+	svc, err := EC2Client(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_vpc_verified_access_group.getVpcVerifiedAccessGroup", "connection_error", err)
+		return nil, err
+	}
+
+	// Build the params
+	input := &ec2.DescribeVerifiedAccessGroupsInput{
+		VerifiedAccessGroupIds: []string{groupId},
+	}
+
+	// Get call
+	op, err := svc.DescribeVerifiedAccessGroups(ctx, input)
+	if err != nil {
+		plugin.Logger(ctx).Error("aws_vpc_verified_access_group.getVpcVerifiedAccessGroup", "api_error", err)
+		return nil, err
+	}
+
+	if op.VerifiedAccessGroups != nil && len(op.VerifiedAccessGroups) > 0 {
```

Comment:
```suggestion
	if op.VerifiedAccessGroups != nil || len(op.VerifiedAccessGroups) > 0 {
```

Do we need both the conditions to be true?

Reply:
Yes, we need both condition to be true, if we will put `||` then `len()` function will throw when `op.VerifiedAccessGroups` value will be nil. `len()` only works on string and array not on `nil`